### ISSUE TITLE
Layer exclusive keyboard focus is not properly released unless the layer is destroyed.

### ISSUE THREAD

**Issue by ItsLemmy** [DreamMaoMao/mangowc#457]
## Info

mango version: 0.10.6
wlroots version: 0.17

## Description

I'm a developer for noctalia-shell and have encountered an issue with how mango handles exclusive keyboard focus on wlr_layer_surfaces. This behavior is currently preventing us from properly implementing mango support in our shell.

When a layer surface's keyboard focus is set from exclusive back to none, the focus is not correctly released, resulting in a permanent loss of keyboard input for all other clients (terminals, applications, etc.).

## Steps to Reproduce
1. Create a full-screen layer surface with keyboard focus initially set to none.
2. Set the layer surface's keyboard focus to exclusive (e.g., when the control panel opens).
3. Set the layer surface's keyboard focus back to none (e.g., when the control panel closes).

Expected Behavior
In step 3, the keyboard focus should be released and returned to the top non-layer surface (e.g., a terminal).

Observed Behavior
In step 3, the focus remains unreleased, leading to a "dangling state" where all keyboard input is permanently lost for other applications.

It appears mango assumes exclusive keyboard focus should only be released when the layer surface is destroyed, which is contrary to the behavior observed in niri, Hyprland, and Sway, were everything works as expected.

## Testing Instructions
This can be easily reproduced by cloning the [noctalia-shell](https://github.com/noctalia-dev/noctalia-shell) repository, checking out the **unified-panel-content**  branch, and running: `qs -p .`

I'm also available for testing here or on your discord.

Thank you!

**Comment by DreamMaoMao** [comment]
fixed in latest commit.

**Comment by ItsLemmy** [comment]
awesome, will test when I wake up tomorrow! ty

**Comment by ItsLemmy** [comment]
Tested and working! tysm
